external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -369,10 +380,17 @@ where
                 Syntax::UnexpectedToken,
                 (loc, format!("Expected {}", item_description))
             );
-            let at_stop_set = skip_comma_list_item(context, start_token, end_token, diag);
-            if at_stop_set {
-                break;
-            }
+            list_error_advance(
+                context,
+                start_token,
+                end_token,
+                /* separator */ Tok::Comma,
+                /* for list */ true,
+                diag,
+            );
+        }
+        if context.at_stop_set() {
+            break;
         }
```

Comment:
Move this to the `while` conditional?

Reply:
We need to attempt parsing a list element before checking for a token to be a part of stop set. In the best case, we will get a less meaningful error message if the item belongs to the token set incorrectly (e.g., `fun` keyword), and in the worst case, we will get an error in the correct code (e.g., if a function argument is named `entry`)

Reply:
That makes sense. Will you add a comment explaining why the logic is ordered this way, so if someone else edits it they will keep the behavior?